src/passes/InstrumentBranchHints.cpp

```diff
@@ -0,0 +1,334 @@
+      getsOfPriorInstrumentation[get] = curr;
+    } else if (auto* tee = curr->operands[2]->dynCast<LocalSet>()) {
+      teesOfPriorInstrumentation[tee] = curr;
+    }
+  }
+
+  bool added = false;
+
+  template<typename T> void processCondition(T* curr) {
+    if (curr->condition->type == Type::unreachable) {
+      // This branch is not even reached.
+      return;
+    }
+
+    auto likely = getFunction()->codeAnnotations[curr].branchLikely;
+    if (!likely) {
+      return;
+    }
+
+    Builder builder(*getModule());
+
+    // Pick an ID for this branch.
+    int id = 0;
+    if (!secondInstrumentation) {
+      // This is the first instrumentation. We instrument everything, using a
+      // new positive ID for each.
+      id = branchId++;
+    } else {
+      // In the second instrumentation we find existing instrumentation and add
+      // paired ones to them. To find the existing ones, we look for this
+      // condition being a local.get that is used in a call to our import, that
+      // is, something like the pattern we emit below:
+      //
+      //  (local.set $temp ..)
+      //  (call LOG_BRANCH (local.get $temp))  ;; used in logging
+      //  (if
+      //    (local.get $temp)                  ;; and used in condition
+      //
+      // We also consider the fallthrough, for the nested case:
+      //
+      //  (if
+      //    (block
+      //      (local.set $temp ..)
+      //      (call LOG_BRANCH (local.get $temp))  ;; used in logging
+      //      (local.get $temp)                    ;; and used in condition
+      //    )
+      //
+      // We also consider a tee:
+      //
+      //  (call LOG_BRANCH (local.tee $temp (..)))  ;; used in logging
+      //  (if
+      //    (local.get $temp)                       ;; and used in condition
+      //
+      auto* fallthrough = Properties::getFallthrough(
+        curr->condition, getPassOptions(), *getModule());
+      auto* get = fallthrough->template dynCast<LocalGet>();
+      if (!get) {
+        return;
+      }
+      auto& sets = localGraph->getSets(get);
+      if (sets.size() != 1) {
+        return;
+      }
+      auto* set = *sets.begin();
+      auto& gets = localGraph->getSetInfluences(set);
+      Call* call = nullptr;
+      if (gets.size() == 2) {
+        // The set has two gets: the get in the condition we began at, and
+        // another.
+        LocalGet* otherGet = nullptr;
+        for (auto* get2 : gets) {
+          if (get2 != get) {
+            otherGet = get2;
+          }
+        }
+        assert(otherGet);
+        // See if that other get is used in a logging.
+        auto iter = getsOfPriorInstrumentation.find(otherGet);
+        if (iter == getsOfPriorInstrumentation.end()) {
+          return;
+        }
+        // Great, this is indeed a prior instrumentation.
+        call = iter->second;
+      } else if (gets.size() == 1) {
+        // The set has only one get, but it might be a tee that flows into a
```

Comment:
```suggestion
        // The set has only one get, but there might be a tee that flows into a
```
The tee is different from the get, right?